Objects/Converters/ConverterRevit/ConverterRevitShared/AllRevitCategories.cs

```diff
@@ -126,32 +145,45 @@ public IRevitCategoryInfo GetRevitCategoryInfo(string categoryName)
     {
       return categoryInfo;
     }
-
+    
     categoryName = CategoryNameFormatted(categoryName);
-    var revitCategoryInfoCache = revitDocumentAggregateCache.GetOrInitializeWithDefaultFactory<IRevitCategoryInfo>();
 
-    categoryInfo = revitCategoryInfoCache.TryGet(categoryName);
-    if (categoryInfo != null)
+    IRevitObjectCache<IRevitCategoryInfo> revitCategoryInfoCache;
+    if (revitDocumentAggregateCache is not null)
     {
-      return categoryInfo;
-    }
+      revitCategoryInfoCache = revitDocumentAggregateCache.GetOrInitializeWithDefaultFactory<IRevitCategoryInfo>();
 
-    foreach (var info in revitCategoryInfoCache.GetAllObjects())
-    {
-      if (categoryName.IndexOf(info.CategoryName, StringComparison.OrdinalIgnoreCase) >= 0)
+      categoryInfo = revitCategoryInfoCache.TryGet(categoryName);
+      if (categoryInfo != null)
       {
-        return info;
+        return categoryInfo;
       }
-      foreach (var alias in info.CategoryAliases)
+      else
       {
-        if (categoryName.IndexOf(alias, StringComparison.OrdinalIgnoreCase) >= 0)
-        {
-          return info;
-        }
+        return SHC.Undefined;
       }
     }
-
-    return SHC.Undefined;
+    else
+    {
+      return SHC.Undefined;
+    }
```

Comment:
could be simpler if we flip the condition? could somewhat reduce nesting and make the logic a bit more linear.

The 2 return statements above this one could also be collapsed into a `return categoryInfo ?? SHC.Undefined` which I consider neater (super personal opinion here...)

Reply:
If this turns into not a 3 second refactor, disregard it.

Reply:
I guess you were referencing to this body (from 156-163) right? Confused because your comment targeting to 166-169

```c#
if (categoryInfo != null)
      {
        return categoryInfo;
      }
      else
      {
        return SHC.Undefined;
      }
```